File: Telemedicine-and-Doctor-Consultation-Platform-main/backend/routes/patients.js
const router = require('express').Router();
const auth   = require('../middleware/auth');
const db     = require('../config/db');

// GET /api/patients
router.get('/', auth, async (req, res) => {
  try {
    const { status, city, q } = req.query;
    let query = 'SELECT * FROM patients WHERE 1=1';
    let params = [];
    
    if (status) { query += ' AND status = ?'; params.push(status); }
    if (city)   { query += ' AND LOWER(city) LIKE ?'; params.push(`%${city.toLowerCase()}%`); }
    if (q)      { query += ' AND LOWER(name) LIKE ?'; params.push(`%${q.toLowerCase()}%`); }
    
    const [rows] = await db.query(query, params);
    
    // To match original frontend payload structure:
    // Some keys might differ marginally (e.g. condition_note vs condition)
    const formattedRows = rows.map(r => ({
      ...r,
      condition: r.condition_note
    }));
      
    res.json({ count: rows.length, data: formattedRows });
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

// GET /api/patients/:id
router.get('/:id', auth, async (req, res) => {
  try {
    const [rows] = await db.query('SELECT * FROM patients WHERE id = ?', [req.params.id]);
    if (!rows.length) return res.status(404).json({ error: 'Patient not found' });
    
    const p = rows[0];
    p.condition = p.condition_note;
    res.json(p);
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

// POST /api/patients
router.post('/', auth, async (req, res) => {
  try {
    const { name, age, city, condition } = req.body;
    if (!name) return res.status(400).json({ error: 'Name required' });
    
    const [countRows] = await db.query('SELECT COUNT(*) as count FROM patients');
    const newId = `P${String(countRows[0].count + 1).padStart(2,'0')}`;
    
    await db.query(
      'INSERT INTO patients (id, name, age, city, condition_note, status) VALUES (?, ?, ?, ?, ?, ?)',
      [newId, name, age || null, city || null, condition || null, 'active']
    );
    
    // Return the new object
    const [newRows] = await db.query('SELECT * FROM patients WHERE id = ?', [newId]);
    res.status(201).json(newRows[0]);
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

// PATCH /api/patients/:id
router.patch('/:id', auth, async (req, res) => {
  try {
    const updates = [];
    const params = [];
    const fields = ['name', 'age', 'city', 'condition_note', 'status'];
    
    for (const key of fields) {
      if (req.body[key] !== undefined) {
        updates.push(`${key} = ?`);
        params.push(req.body[key]);
      }
    }
    
    // Alternative mapping for 'condition'
    if (req.body.condition !== undefined) {
       updates.push(`condition_note = ?`);
       params.push(req.body.condition);
    }
    
    if (updates.length > 0) {
      params.push(req.params.id);
      await db.query(`UPDATE patients SET ${updates.join(', ')} WHERE id = ?`, params);
    }
    
    const [rows] = await db.query('SELECT * FROM patients WHERE id = ?', [req.params.id]);
    if (!rows.length) return res.status(404).json({ error: 'Patient not found' });
    
    res.json(rows[0]);
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

module.exports = router;

File: Telemedicine-and-Doctor-Consultation-Platform-main/backend/routes/patients.py
from flask import Blueprint, request, jsonify
from config import get_db_connection
from middleware.auth import token_required

bp = Blueprint('patients', __name__, url_prefix='/api/patients')

@bp.route('/', methods=['GET'])
@token_required
def get_patients():
    try:
        status = request.args.get('status')
        city = request.args.get('city')
        q = request.args.get('q')
        
        query = 'SELECT * FROM patients WHERE 1=1'
        params = []
        
        if status:
            query += ' AND status = %s'
            params.append(status)
        if city:
            query += ' AND LOWER(city) LIKE %s'
            params.append(f'%{city.lower()}%')
        if q:
            query += ' AND LOWER(name) LIKE %s'
            params.append(f'%{q.lower()}%')
        
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, params)
        rows = cursor.fetchall()
        cursor.close()
        conn.close()
        
        # Format response
        formatted = [dict(r, condition=r.get('condition_note')) for r in rows]
        
        return jsonify({'count': len(rows), 'data': formatted}), 200
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500

@bp.route('/<id>', methods=['GET'])
@token_required
def get_patient(id):
    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute('SELECT * FROM patients WHERE id = %s', (id,))
        patient = cursor.fetchone()
        cursor.close()
        conn.close()
        
        if not patient:
            return jsonify({'error': 'Patient not found'}), 404
        
        patient['condition'] = patient.get('condition_note')
        return jsonify(patient), 200
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500

@bp.route('/', methods=['POST'])
@token_required
def create_patient():
    try:
        data = request.get_json()
        name = data.get('name')
        age = data.get('age')
        city = data.get('city')
        condition = data.get('condition')
        
        if not name:
            return jsonify({'error': 'Name required'}), 400
        
        conn = get_db_connection()
        cursor = conn.cursor()
        
        # Get count for ID
        cursor.execute('SELECT COUNT(*) as count FROM patients')
        result = cursor.fetchone()
        count = result['count'] + 1
        new_id = f'P{str(count).zfill(2)}'
        
        # Insert patient
        cursor.execute(
            'INSERT INTO patients (id, name, age, city, condition_note, status) VALUES (%s, %s, %s, %s, %s, %s)',
            (new_id, name, age, city, condition, 'active')
        )
        conn.commit()
        
        # Get the created patient
        cursor.execute('SELECT * FROM patients WHERE id = %s', (new_id,))
        patient = cursor.fetchone()
        cursor.close()
        conn.close()
        
        return jsonify(patient), 201
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500

@bp.route('/<id>', methods=['PATCH'])
@token_required
def update_patient(id):
    try:
        data = request.get_json()
        updates = []
        params = []
        
        fields = ['name', 'age', 'city', 'condition_note', 'status']
        
        for field in fields:
            if field in data and data[field] is not None:
                updates.append(f'{field} = %s')
                params.append(data[field])
        
        # Handle alternative mapping for 'condition'
        if 'condition' in data and data['condition'] is not None:
            if 'condition_note = %s' not in updates:
                updates.append('condition_note = %s')
                params.append(data['condition'])
        
        if not updates:
            return jsonify({'error': 'Nothing to update'}), 400
        
        params.append(id)
        
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(f"UPDATE patients SET {', '.join(updates)} WHERE id = %s", params)
        conn.commit()
        
        cursor.execute('SELECT * FROM patients WHERE id = %s', (id,))
        patient = cursor.fetchone()
        cursor.close()
        conn.close()
        
        if not patient:
            return jsonify({'error': 'Patient not found'}), 404
        
        return jsonify(patient), 200
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500
